Fails to compile on Asterisk 16

Asterisk 16 has more restrictions on code calls, thus the module fails to build against 16
```
cc -c -I../asterisk-amqp -DHAVE_STDINT_H=1 -Wall -Wextra -Wno-unused-parameter -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -Winit-self -Wmissing-format-attribute -Wformat=2 -g -fPIC -D_GNU_SOURCE -D'AST_MODULE="res_stasis_amqp"' -D'AST_MODULE_SELF_SYM=__internal_res_stasis_amqp_self' -o res_stasis_amqp.o res_stasis_amqp.c
In file included from /usr/include/asterisk.h:23:0,
                 from res_stasis_amqp.c:71:
res_stasis_amqp.c: In function '_dtor_routing_key':
res_stasis_amqp.c:337:38: error: 'Do_not_use_free__use_ast_free_or_ast_std_free_for_remotely_allocated_memory' undeclared (first use in this function)
  RAII_VAR(char *, routing_key, NULL, free);
                                      ^
res_stasis_amqp.c:337:38: note: each undeclared identifier is reported only once for each function it appears in
res_stasis_amqp.c: In function 'new_routing_key':
res_stasis_amqp.c:385:22: error: 'Do_not_use_malloc__use_ast_malloc' undeclared (first use in this function)
  if (!(routing_key = malloc(routing_key_len + 1))) {
                      ^
res_stasis_amqp.c: In function '_dtor_routing_key':
res_stasis_amqp.c:410:38: error: 'Do_not_use_free__use_ast_free_or_ast_std_free_for_remotely_allocated_memory' undeclared (first use in this function)
  RAII_VAR(char *, routing_key, NULL, free);
                                      ^
res_stasis_amqp.c: In function '_dtor_routing_key':
res_stasis_amqp.c:578:38: error: 'Do_not_use_free__use_ast_free_or_ast_std_free_for_remotely_allocated_memory' undeclared (first use in this function)
  RAII_VAR(char *, routing_key, NULL, free);
                                      ^
In file included from res_stasis_amqp.c:73:0:
res_stasis_amqp.c: At top level:
res_stasis_amqp.c:676:1: error: unknown field 'nonoptreq' specified in initializer
 AST_MODULE_INFO(ASTERISK_GPL_KEY, AST_MODFLAG_DEFAULT, "Send all Stasis messages to AMQP",
```

--- a/res_stasis_amqp.c
+++ b/res_stasis_amqp.c
@@ -334,7 +334,7 @@ static void send_ami_event_to_amqp(void *data, struct stasis_subscription *sub,
 									struct stasis_message *message)
 {
 	RAII_VAR(struct ast_json *, json, NULL, ast_json_unref);
-	RAII_VAR(char *, routing_key, NULL, free);
+	RAII_VAR(char *, routing_key, NULL, ast_free);
 	RAII_VAR(struct ast_manager_event_blob *, manager_blob, NULL, ao2_cleanup);
 	const char *routing_key_prefix = "stasis.ami";
 	int res = 0;
@@ -382,7 +382,7 @@ char *new_routing_key(const char *prefix, const char *suffix)
 		*ptr = tolower(*ptr);
 	}
 
-	if (!(routing_key = malloc(routing_key_len + 1))) {
+	if (!(routing_key = ast_malloc(routing_key_len + 1))) {
 		ast_log(LOG_ERROR, "failed to allocate a string for the routing key\n");
 		return NULL;
 	}
@@ -407,7 +407,7 @@ static int stasis_amqp_channel_log(struct stasis_message *message)
 	RAII_VAR(struct ast_json *, json, NULL, ast_json_free);
 	RAII_VAR(struct ast_json *, channel, NULL, ast_json_free);
 	RAII_VAR(struct ast_json *, unique_id, NULL, ast_json_free);
-	RAII_VAR(char *, routing_key, NULL, free);
+	RAII_VAR(char *, routing_key, NULL, ast_free);
 	const char *routing_key_prefix = "stasis.channel";
 
 	if (!(json = stasis_message_to_json(message, NULL))) {
@@ -575,7 +575,7 @@ static int unload_module(void)
 
 static void stasis_app_message_handler(void *data, const char *app_name, struct ast_json *message)
 {
-	RAII_VAR(char *, routing_key, NULL, free);
+	RAII_VAR(char *, routing_key, NULL, ast_free);
 	const char *routing_key_prefix = "stasis.app";
 
 	if (!(routing_key = new_routing_key(routing_key_prefix, app_name))) {
@@ -676,6 +676,5 @@ static int load_module(void)
 AST_MODULE_INFO(ASTERISK_GPL_KEY, AST_MODFLAG_DEFAULT, "Send all Stasis messages to AMQP",
 	.support_level = AST_MODULE_SUPPORT_EXTENDED,
 	.load = load_module,
-	.unload = unload_module,
-	.nonoptreq = "res_stasis_amqp"
+	.unload = unload_module
 );

